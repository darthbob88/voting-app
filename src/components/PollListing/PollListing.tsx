import React, { useState } from 'react';
import { retrievePollList } from '../../model/DataAccess';
import { Poll, testPoll } from '../../model/Poll';

import styles from "./PollListing.module.css";

function PollListing() {
    const pollList = retrievePollList();
    return (
        <div>
            <h2>List of Polls</h2>
            <ul>
                {pollList.map(poll => <li key={poll} ><a href={poll}>{poll}</a> </li>)}
            </ul>
        </div>
    );
}

export default PollListing;