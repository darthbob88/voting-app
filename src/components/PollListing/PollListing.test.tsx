import { render } from '@testing-library/react';
import PollListing from './PollListing';

test('renders PollListing component', () => {
    const { getByText, container } = render(<PollListing />);
    expect(container).toMatchSnapshot();
});
