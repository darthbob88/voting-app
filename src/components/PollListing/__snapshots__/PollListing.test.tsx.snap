// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PollListing component 1`] = `
<div>
  <div>
    <h2>
      List of Polls
    </h2>
    <ul />
  </div>
</div>
`;
